src/Symfony/Component/Amqp/Queue.php

```diff
@@ -0,0 +1,142 @@
+        } else {
+            $routingKeys = array($name);
+        }
+
+        if (isset($arguments['flags'])) {
+            $this->setFlags($arguments['flags']);
+            unset($arguments['flags']);
+        } else {
+            $this->setFlags(\AMQP_DURABLE);
+        }
+
+        if (isset($arguments['exchange'])) {
+            $exchange = $arguments['exchange'];
+            unset($arguments['exchange']);
+        } else {
+            $exchange = Broker::DEFAULT_EXCHANGE;
+        }
+
+        if (array_key_exists('retry_strategy', $arguments)) {
+            $this->retryStrategy = $arguments['retry_strategy'];
+            if (!$this->retryStrategy instanceof RetryStrategyInterface) {
+                throw new \InvalidArgumentException('The retry_strategy should be an instance of RetryStrategyInterface.');
+            }
+            unset($arguments['retry_strategy']);
+        }
+
+        if (array_key_exists('retry_strategy_queue_pattern', $arguments)) {
+            $this->retryStrategyQueuePattern = $arguments['retry_strategy_queue_pattern'];
+            unset($arguments['retry_strategy_queue_pattern']);
+        } else {
+            $this->retryStrategyQueuePattern = '%exchange%.%time%.wait';
+        }
+
+        if (isset($arguments['bind_arguments'])) {
+            $bindArguments = $arguments['bind_arguments'];
+            unset($arguments['bind_arguments']);
+        } else {
+            $bindArguments = array();
+        }
+
+        $this->setArguments($arguments);
+
+        if (null === $routingKeys) {
+            $this->bindings[$exchange][] = array(
+                'routing_key' => $routingKeys,
+                'bind_arguments' => $bindArguments,
+            );
+        } elseif (is_array($routingKeys)) {
+            foreach ($routingKeys as $routingKey) {
+                $this->bindings[$exchange][] = array(
+                    'routing_key' => $routingKey,
+                    'bind_arguments' => $bindArguments,
+                );
+            }
+        }
+
+        // Special binding: Bind this queue, with its name as the routing key
+        // with the retry exchange in order to have a nice retry workflow.
+        $this->bindings[Broker::RETRY_EXCHANGE][] = array(
+            'routing_key' => $name,
+            'bind_arguments' => array(),
+        );
+
+        if ($declare) {
+            $this->declare();
+        }
+    }
+
+    public function declare()
+    {
+        $this->declareQueue();
+
+        foreach ($this->bindings as $exchange => $configs) {
+            foreach ($configs as $config) {
+                parent::bind($exchange, $config['routing_key'], $config['bind_arguments']);
+            }
+        }
+    }
+
+    public function getBindings()
+    {
+        return $this->bindings;
+    }
+
+    public function getRetryStrategy()
+    {
+        return $this->retryStrategy;
+    }
+
+    public function getRetryStrategyQueuePattern()
```

Comment:
I'm not too familiar with AMQP, but it looks like these methods are new, they should get some documentation, if any, a return type